lib/src/neu/parsers/headers-parser.ts

```diff
@@ -14,28 +15,33 @@ export function parseHeaders(
   parameter: ParameterDeclaration,
   typeTable: TypeTable,
   lociTable: LociTable
-): Header[] {
+): Result<Header[], ParserError> {
   parameter.getDecoratorOrThrow("headers");
-  // TODO check parameter.isOptional()
   if (parameter.hasQuestionToken()) {
-    throw new OptionalNotAllowedError("@headers parameter cannot be optional", {
-      file: parameter.getSourceFile().getFilePath(),
-      position: parameter.getQuestionTokenNodeOrThrow().getPos()
-    });
+    return err(
+      new OptionalNotAllowedError("@headers parameter cannot be optional", {
+        file: parameter.getSourceFile().getFilePath(),
+        position: parameter.getQuestionTokenNodeOrThrow().getPos()
+      })
+    );
   }
   const type = getParameterTypeAsTypeLiteralOrThrow(parameter);
-  const headers = type
-    .getProperties()
-    .map(p => {
-      const pDescription = getJsDoc(p);
-      return {
-        name: getPropertyName(p),
-        type: parseType(p.getTypeNodeOrThrow(), typeTable, lociTable),
-        description: pDescription && pDescription.getComment(),
-        optional: p.hasQuestionToken()
-      };
-    })
-    .sort((a, b) => (b.name > a.name ? -1 : 1));
+
+  const headers = [];
+  for (const ps of type.getProperties()) {
```

Comment:
what's ps? Post script?

Reply:
property signature - will update var to `propertySignature`